### ISSUE TITLE
bug: data structures compilation error (merkle_tree.cairo:57:46)

### ISSUE THREAD

**Issue by 0xChqrles** [keep-starknet-strange/alexandria#107]
# Bug Report

**Alexandria version:**

df9ed6cda0288c2f05308006e1e9230a564123dc

**Current behavior:**

running `scarb build` in `alexandria/data_structures` throw this error

```
error: Type annotations needed. Failed to infer ?5
 --> merkle_tree.cairo:57:46
                    if current_node.into() < (*proof_element).into() {
                                             ^*********************^
```

**Expected behavior:**

Should compile

**Steps to reproduce:**

Install the following version of scarb:

```
scarb 0.3.0 (aa281bcc0 2023-06-02)
cairo: 1.1.0 (43b83560d)
```

run `scarb build` in `alexandria/data_structures`

**Comment by lambda-0x** [comment]
because its fixed now, can you close this @0xChqrles?

**Comment by LucasLvy** [comment]
Fixed by #106
